ConfigSpace/configuration_space.py

```diff
@@ -113,9 +116,10 @@ def add_hyperparameter(self, hyperparameter):
         self._check_default_configuration()
         self._sort_hyperparameters()
 
+        # todo: check it returns input as aoutput?
         return hyperparameter
 
-    def add_condition(self, condition):
+    def add_condition(self, condition: ConditionComponent) -> ConditionComponent:
```

Comment:
You're right, ConditionComponent is a parent class of AbstractCondition; accepting and returning a ConditionComponent is totally correct.